docker/docker_image_test.go

```diff
@@ -0,0 +1,109 @@
+			resp.Header.Add("Content-Length", fmt.Sprintf("%d", len(body)))
+			resp.Header.Add("Strict-Transport-Security", "max-age=31536000")
+			resp.Body = io.NopCloser(bytes.NewReader(body))
+			resp.StatusCode = http.StatusUnauthorized
+			resp.Status = http.StatusText(resp.StatusCode)
+		} else if r.Method == http.MethodGet && regexp.MustCompile("https://auth.docker.io/token?.*scope=repository%3Alibrary%2Fbusybox%3Apull&service=registry.docker.io").MatchString(url) {
+			resp.Header = http.Header{}
+			resp.Header.Add("Content-Type", "application/json; charset=utf-8")
+			resp.Header.Add("X-Trace-Id", "c00a2a131c306bbe24032054e4b958fb")
+			resp.Header.Add("Strict-Transport-Security", "max-age=31536000")
+			resp.Body = io.NopCloser(bytes.NewReader([]byte(tokenResp)))
+			resp.StatusCode = http.StatusOK
+			resp.Status = http.StatusText(resp.StatusCode)
+		} else if r.Method == http.MethodHead && url == "https://registry-1.docker.io/v2/library/busybox/manifests/1.34.1" {
+			resp.Header = http.Header{}
+			resp.Header.Add("Content-Length", "2295")
+			resp.Header.Add("Docker-Content-Digest", "sha256:05a79c7279f71f86a2a0d05eb72fcb56ea36139150f0a75cd87e80a4272e4e39")
+			resp.Header.Add("Etag", `"sha256:05a79c7279f71f86a2a0d05eb72fcb56ea36139150f0a75cd87e80a4272e4e39"`)
+			resp.Header.Add("Ratelimit-Remaining", "0;w=21600")
+			resp.Header.Add("Docker-Ratelimit-Source", "131.228.2.25")
+			resp.Header.Add("Content-Type", internalManifest.DockerV2ListMediaType)
+			resp.Header.Add("Docker-Distribution-Api-Version", "registry/2.0")
+			resp.Header.Add("Strict-Transport-Security", "max-age=31536000")
+			resp.Header.Add("Ratelimit-Limit", "100;w=21600")
+			resp.ContentLength = 2295
+			resp.StatusCode = http.StatusOK
+			resp.Status = http.StatusText(resp.StatusCode)
+		} else {
+			return nil, fmt.Errorf("%w: %s %s", errFakeResponseNotFound, r.Method, url)
+		}
+
+		return resp, nil
+	}))
+
+	imageRef, err := ParseReference("//docker.io/library/busybox:1.34.1")
+	require.NoError(t, err)
+	digest, err := GetDigest(context.Background(), sys, imageRef)
+	assert.NoError(t, err)
+	assert.Equal(t, "sha256:05a79c7279f71f86a2a0d05eb72fcb56ea36139150f0a75cd87e80a4272e4e39", string(digest))
+}
+
+func setupTestCase(t *testing.T) (string, *types.SystemContext) {
+	tempHome := t.TempDir()
+	rootDirPath := filepath.Join(tempHome, "root")
+	err := os.MkdirAll(rootDirPath, 0700)
+	require.NoError(t, err)
+	userRegistriesDirPath := filepath.Join(tempHome, filepath.FromSlash(".config/containers/registries.d"))
+	err = os.MkdirAll(userRegistriesDirPath, 0700)
+	require.NoError(t, err)
+	bigFilesDirPath := filepath.Join(rootDirPath, "containers", "bigcache")
+	err = os.MkdirAll(bigFilesDirPath, 0700)
+	require.NoError(t, err)
+
+	return tempHome, &types.SystemContext{
+		RootForImplicitAbsolutePaths:      rootDirPath,
+		RegistriesDirPath:                 userRegistriesDirPath,
```

Comment:
(Setting this to `"/this/does/not/exist"` works well enough for a unit test, without having to actually create a directory.)